### PR TITLE
add GaussianDropout, GaussianNoise and unit tests, python wrapper

### PR DESCRIPTION
**Pull request by shane-huang**
## What changes were proposed in this pull request?

Added 2 noise layers for Keras:  GaussianDropout, GaussianNoise, with unit tests, python wrapper, serializatoin test.. 

## How was this patch tested?

PR validation test passed. 

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/XXX

Do we need compare with Keras? How to deal with random numbers?

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/python/api/PythonBigDL.scala [hunk 1]
```diff
@@ -527,6 +527,16 @@ class PythonBigDL[T: ClassTag](implicit ev: TensorNumeric[T]) extends Serializab
     Dropout[T](initP, inplace, scale)
   }
 
+  def createGaussianDropout(rate: Double)
```
Comment (zhichao-li) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/python/api/PythonBigDL.scala]: Expose this layer in python. please refer to `Linear` in "layer.py"

**Comment by zhichao-li** [comment]
docs for bigdl-project.github.io ?

**Comment by wzhongyuan** [comment]
LGTM

**Comment by zhichao-li** [comment]
Wondering do we need to differentiate batch or non-batch input and add doc to specify the input format?
LGTM except for this minor one.

**Pull request merged**